packages/core/test/types.test.ts

```diff
@@ -4339,6 +4339,24 @@ describe('self', () => {
       })
     });
   });
+
+  it('should allow actor union snap snapshot methods to be called', () => {
```

Comment:
This was added to the `describe` block that is about `self`. Could you create a new `describe` block and move this test into it?

Reply:
Improved the tests as well. Something I learned that may or may not be intentional. If you don't call setup, the `snapshot.matches` method parameters don't infer the different possible machine states.

Reply:
> If you don't call setup, the snapshot.matches method parameters don't infer the different possible machine states.

Supporting this in the bare `createMachine` would make us require TS 5.3 (which is the current latest version) so we decided not to do that there. The problem is that it would have negative effects on users of other TS versions.